components/Statistics/TourStats.tsx

```diff
@@ -84,28 +73,60 @@ export default function TourStats(props: TourStatsProps) {
           <View style={styles.stat_row}>
             <Icon name="calendar-plus" style={styles.icon_style} />
             <Text>
-              {props.tour.startedAt
-                ? formatDate(props.tour.startedAt, DateFormat.DATE)
+              {activeTour!.startedAt
+                ? formatDate(activeTour!.startedAt, DateFormat.DATE)
                 : "--"}
             </Text>
           </View>
-          {props.tour.finishedAt && (
+          {activeTour!.finishedAt && (
             <View style={styles.stat_row}>
               <Icon name="calendar-check" style={styles.icon_style} />
-              <Text>{formatDate(props.tour.finishedAt, DateFormat.DATE)}</Text>
+              <Text>{formatDate(activeTour!.finishedAt, DateFormat.DATE)}</Text>
             </View>
           )}
           <View style={styles.stat_row}>
             <Icon name="clock" style={styles.icon_style} />
-            <Text>{getTotalMillisecondsString(getTourDuration(stages))}</Text>
+            <Text>
+              {getTotalMillisecondsString(getTourDuration(tourStages))}
+            </Text>
           </View>
         </View>
       </View>
     </View>
   );
 }
 
-const makeStyles = (theme: ThemeType): any => {
+// observe the stage (tracks updates to the stage)
+const enhance = withObservables(["stage"], ({ stage }: { stage: Stage }) => ({
+  stage,
+}));
+
+const EnhancedTourStats = enhance(TourStats);
+
+// Bridge component that determines which TourStats component to render based on the active stage.
+const Bridge = ({ stages }: { stages: Stage[] }) => {
+  tourStages = stages;
+  stages.forEach((stage) => {
+    if (stage.isActive) {
+      return <EnhancedTourStats stage={stage} />;
```

Comment:
Ich glaube das return hier returnt nur innerhalb des Lambdas, returnt aber nicht die <EnhancedTourStats> in der Funktion Bridge selbst

Reply:
Wenn man das ganz mit dem Element Inspector anschaut, dann sieht man auch, das selbst bei aktiver Stage die normale TourStat geladen wird
<img width="528" alt="Bildschirmfoto 2024-12-17 um 15 18 22" src="https://github.com/user-attachments/assets/9580a9b6-512f-4c57-a6d7-2d96dddc1dd3" />

Reply:
Scheint zu funktionieren, sehr gut